oauthproxy.go

```diff
@@ -1248,3 +1251,27 @@ func (p *OAuthProxy) errorJSON(rw http.ResponseWriter, code int) {
 	// application/json
 	rw.Write([]byte("{}"))
 }
+
+// LoggingCSRFCookiesInOAuthCallback Log all CSRF cookies found in HTTP request OAuth callback,
+// which were successfully parsed
+func LoggingCSRFCookiesInOAuthCallback(req *http.Request, cookieName string) {
+	cookies := req.Cookies()
+	if len(cookies) == 0 {
+		logger.Println(req, logger.AuthFailure, "No cookies were found in OAuth callback.")
+	} else {
+		var anyFound = false
+		for i := range cookies {
+			var c = cookies[i]
+			if cookieName == c.Name {
+				logger.Println(req, logger.AuthFailure, "The CSRF cookie %s was found in OAuth callback.", c.Name)
+				anyFound = true
+			} else if strings.Contains(c.Name, "_csrf") {
+				logger.Println(req, logger.AuthFailure, "The CSRF cookie %s was found in OAuth callback, but it is not the expected one (%s).", c.Name, cookieName)
+				anyFound = true
+			}
+		}
+		if !anyFound {
```

Comment:
Leave an empty new line before this statement to separate logical blocks of the code flow